indexer/ai/src/main/java/com/google/cloudsearch/ai/CustomSkill.java

```diff
@@ -0,0 +1,200 @@
+   *                    information.
+   * @param schema      The CloudSearch Schema.
+   * @throws InvalidConfigException
+   */
+    public CustomSkill(JSONObject aiSkill, JSONObject schema) throws InvalidConfigException {
+            this.parse(aiSkill, schema);
+    }
+
+    /**
+     * Set the Inputs for the Cloud Function
+     * @param input     JSON Input Object
+     */
+    @Override
+    public void setInputs(JSONObject input) {
+        this.input = input;
+    }
+
+    /**
+     * Return the Inputs
+     * @return     Returns the Inputs for the skill in JSON format.
+     */
+    @Override
+    public JSONObject getInputs() {
+        return this.input;
+    }
+
+    /**
+     * Parse the JSON Object containing the inputs for the skill.
+     * @param input     Input in JSON format.
+     * @throws InvalidConfigException   Throws exception if the key in the input object is not expected.
+     */
+    @Override
+    public void parseInputs(JSONObject input) throws InvalidConfigException {
+        if(input == null) {
+            setInputs(new JSONObject());
+        }
+        else
+            setInputs(input);
+    }
+
+    /**
+     * No Filters are required for the Custom Skill.
+     * @param filter
+     */
+    @Override
+    public void setFilter(JSONObject filter) { return; }
+
+    /**
+     * Returns the filters.
+     * @return
+     */
+    @Override
+    public JSONObject getFilter() {
+        return null;
+    }
+
+    /**
+     * Filters not required for Custom AI Skill.
+     * @param filter
+     * @throws InvalidConfigException
+     */
+    @Override
+    public void parseFilter(JSONObject filter) throws InvalidConfigException {
+        if(filter != null) {
+            throw new InvalidConfigException("Filters not supported in Custom AI Skill.");
+        }
+    }
+
+    /**
+     * Set the Cloud Function URL.
+     * @param url   URL for invoking the Cloud Function.
+     */
+    public void setURL(String url) {
+        this.url = url;
+    }
+
+    /**
+     * Return the Cloud Function URL
+     * @return
+     */
+    public String getURL() {
+        return this.url;
+    }
+
+    /**
+     * Parses the Configuration for the Custom Skill.
+     * @param aiSkill        JSONObject for the skill
+     * @param schema         CloudSearch schema
+     * @throws InvalidConfigException
+     */
+    @Override
+    protected void parse(JSONObject aiSkill, JSONObject schema) throws InvalidConfigException {
+        parseAISkillName((String) aiSkill.get(Constants.CONFIG_SKILL_NAME));
+        parseOutputMappings((JSONArray) aiSkill.get(Constants.CONFIG_OUTPUT_MAPPINGS), schema);
+        parseInputs((JSONObject) aiSkill.get(Constants.CONFIG_INPUTS));
+        parseFilter((JSONObject) aiSkill.get(Constants.CONFIG_FILTERS));
+        setURL((String) aiSkill.get(Constants.CONFIG_CLOUD_FUNCTION_URL));
+    }
+
+    /**
+     *  No specific setup required.
+     */
+    @Override
+    public void setupSkill() {}
+
```

Comment:
why aren't we establishing the connection to cloud function in setup ?

Reply:
done